### ISSUE TITLE
Extended Display and 2 GPU's

### ISSUE THREAD

**Issue by dimipisi** [ledoge/dwm_lut#24]
@ledoge First of all I want to thank you for your work and for this product. It is trully a great tool and it saved me the trouble and money because I was set on buying a new monitor before finding this tool!

Now, the problem I encountered. My laptop has 2 GPU's. An integrated one and a dedicated one. My laptop is always in hybrid mode because I usually use my laptop on different places and I need the battery life when I'm on the go.
If the laptop is in hybrid mode, when I use both monitors at the same time (in extended mode), both monitors are assigned the same number (1) in DWM_LUT. This causes the program to assign the lut but a few seconds later the screen goes black and then the program goes to the state of "inactive".
If I disable hybrid mode (dedicated gpu only), the program assigns different numbers to the displays, Laptop display number one and external display number 2, and the luts are applied.

Maybe a simple fix would be to not allow the program to assign the same number to different monitors. If a number is already assigned, then the program assigns a different number.

![iGPU + dGPU](https://user-images.githubusercontent.com/105156650/167301909-f8bb1bb4-6aaf-45da-823f-ea0bc942bfa1.jpg)
![dGPU](https://user-images.githubusercontent.com/105156650/167301925-7396d005-b42e-4b14-bcb2-916d0ea68b2e.jpg)

**Comment by ledoge** [comment]
Thanks!

Very interesting issue. The number shown is just used in the GUI, not internally, so that's not the cause of the issue. However, that value, as reported by Windows, *should* normally be unique, so there's definitely something strange going on behind the scenes. If I had to guess, maybe the Desktop is actually rendered on the dedicated GPU for the external monitor? In that case, only having a LUT assigned to one of the monitors should work fine – please let me know if that's right. Unfortunately, as I don't have a Laptop with a dedicated GPU, I can't really troubleshoot and fix the crash. If my guess is right, it would require a fair amount of changes to the code, so not feasible unless I can actually test it locally.

**Comment by dimipisi** [comment]
Yeah, the external display is controlled by the dedicated GPU, so I assume the issue is probably related to the process being applied on two different gpu's at the same time. I'll try and only apply a lut to one display when in extended mode, and tomorrow i'll tell you the result.

Thanks!

**Comment by dimipisi** [comment]
@ledoge So, I tried to apply only one lut to one monitor and here are the results.

If I apply only one lut to the laptop display, it fails.
But if I only apply the lut to the external display, it works.

But then I remembered something. I went to the windows settings and switched my laptop screen to be the main display. Now DWM_LUT only stays active if I apply the lut only to the laptop display.
So, it works only on the main display selected in the windows settings.

Running Win10.
